docs/source/format/Columnar.rst

```diff
@@ -487,6 +498,103 @@ will be represented as follows: ::
           |-------------------------------|-----------------------|
           | 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 | unspecified (padding) |
 
+ListView Layout
+~~~~~~~~~~~~~~~
+
+The ListView layout is defined by three buffers instead of just two:
+a validity bitmap, an offsets buffer, and an additional sizes buffer.
+The sizes have the same bit width as the offsets and both 32-bit and 64-bit
+signed integer options are supported. Like in the List layout, the offsets
+reference the child array.
+
+Rather then inferring list lengths from the offsets, the sizes buffer
+stores the length of each list in the array. This in turn allows offsets to be
```

Comment:
```suggestion
Sizes and offsets have the identical bit width and both 32-bit and 64-bit
signed integer options are supported.

As in the List layout, the offsets encode the start position of each slot
in the child array. In contrast to the List layout, list lengths are stored
explicitly in the sizes buffer instead of inferred. This allows offsets to be
```